pandas/tests/series/test_arithmetic.py

```diff
@@ -82,3 +82,109 @@ def test_pow_ops_object():
     expected = Series(b.values ** a.values, dtype=object)
 
     tm.assert_series_equal(result, expected)
+
+
+# ------------------------------------------------------------------
+# Unary
+
+class TestSeriesUnary(object):
+
+    @pytest.mark.parametrize('typ', ['int64', 'int32', 'int16', 'int8',
+                                     'uint64', 'uint32', 'uint16', 'uint8',
+                                     'Int64', 'Int32', 'Int16', 'Int8',
+                                     'UInt64', 'UInt32', 'UInt16', 'UInt8'])
+    @pytest.mark.parametrize('op', [operator.pos, operator.neg,
+                                    operator.inv, operator.abs])
+    def test_integer(self, typ, op):
+        # GH#23087
+        ser = Series([1, 3, 2], index=range(3), dtype=typ)
+        result = op(ser)
+        exp = Series(op(ser.values), index=ser.index)
+        tm.assert_series_equal(result, exp)
+
+    @pytest.mark.parametrize('typ', ['float64', 'float32', 'float16'])
+    @pytest.mark.parametrize('op', [operator.pos, operator.neg,
+                                    operator.inv, operator.abs])
+    def test_float(self, typ, op):
+        # GH#23087
+        ser = Series([1.1, 3.1, 2.1], index=range(3), dtype=typ)
+
+        if op is operator.inv:
```

Comment:
simpler to do operator.inv in a separate test and leave it out of the parameterize